### DISCUSSION TITLE
RTI1W - Minor comment/question

### DISCUSSION THREAD

**Discussion by OptimisticMonkey** [RayTracing/raytracing.github.io#888] [Q&A]
In the beginning section that only renders a background ( listing 9)... I have a very minor clarification/comment...

It is implied that y component of the unit direction of the ray will vary from -1 to 1 when rendering the image.

As I understand it - this is the direction from the origin to the top and bottom of the box, and is a smaller interval...
eg like -.7 to .7.

So the background generated by
```
color ray_color(const ray& r) {
    vec3 unit_direction = unit_vector(r.direction());
    auto t = 0.5*(unit_direction.y() + 1.0);
    return (1.0-t)*color(1.0, 1.0, 1.0) + t*color(0.5, 0.7, 1.0);
}
```
Doesnt really interpolate between 1.0, 1.0, 1.0) to (0.5, 0.7, 1.0) but rather a smaller range - 

Is that a correct understanding?

**Comment by hollasch** [comment]
It does interpolate from (1, 1, 1) to (.5, .7, 1), though you won't see the full range using the given viewport. Pure white straight down, and pure (.5, .7, 1) when looking straight up. Think of it as specifying the color of the surrounding environment, rather than coloring any specific viewport.

**Reply by OptimisticMonkey** [reply]
Thanks for replying Hollasch!  

I think I get what you are saying - there exists an alternate possible viewport configuration where you are "almost looking straight up" and unit_direction.y() would approach very close to 1.  (eg the camera plane is enormously huge and right next to the eye)

Just fired up debugger and verified the default setup in the book unit_direction.y() only goes -0.7 to 0.7.

Thanks again for replying - helped me think about it further
